lib/spack/spack/relocate.py

```diff
@@ -88,7 +88,8 @@ def _patchelf():
         return patchelf.path
 
     # Check if patchelf spec is installed
-    spec = spack.spec.Spec('patchelf').concretized()
+    spec = spack.spec.Spec('patchelf')
+    spec._old_concretize()
```

Comment:
Why do we explicitly use the old concretizer for patchelf? Is that because we rely on it to install the clingo binaries?

If so, I think we should do something more radical here, in the interests of being able to eventually remove the old concretizer entirely. We should construct a concrete spec for patchelf synthetically.

Since patchelf has no dependencies, this is pretty easy -- we can take a known concrete spec for patchelf, read it in from yaml, swap the compiler  to something we know we have, swap the architecture to the current architecture, and install it.

Reply:
> Why do we explicitly use the old concretizer for patchelf? Is that because we rely on it to install the clingo binaries?

Yes, it's because we need it to relocate clingo - so if it's not already installed in the system we have to concretize it. I agree with what you say, and I have on my agenda to check if we need to do anything at all to relocate `patchelf` itself. In case we do not,  we can also provide a binary that just needs to be untarred and copied to location. 

Do you want to take care of `patchelf` in this PR or would it be fine to have a subsequent PR focused only on that?

Reply:
I think it's fine to do that in a follow-on PR